agent/internal/taskoutput/directory.go

```diff
@@ -107,4 +109,4 @@ type directoryHandler interface {
 }
 
 // directoryHandlerFactory abstracts the creation of a directory handler.
-type directoryHandlerFactory func(string, *taskoutput.TaskOutput, taskoutput.TaskOptions, redactor.RedactionOptions, client.LoggerProducer) directoryHandler
+type directoryHandlerFactory func(string, *taskoutput.TaskOutput, taskoutput.TaskOptions, redactor.RedactionOptions, client.LoggerProducer, *grpc.ClientConn) directoryHandler
```

Comment:
I am rethinking this (and I probably should have done this originally) but the factory function signature should probably be:

```
type directoryHandlerFactory func(string, *taskoutput.Taskoutput, taskoutputOptions, client.LoggerProducer, interface{})
```

where `interface{}` is the factory specific options struct.

Reply:
I created a `directoryHandlerOpts` that gets passed into both factories. Let me know what you think.